pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java

```diff
@@ -2579,7 +2580,15 @@ public String getBookkeeperMetadataStoreUrl() {
         } else {
             // Fallback to same metadata service used by broker, adding the "metadata-store" to specify the BK
             // metadata adapter
-            return "metadata-store:" + getMetadataStoreUrl();
+            String suffix;
+            if (StringUtils.isNotBlank(metadataStoreUrl)) {
```

Comment:
wouldn't we have the same issue here if a user configure `metadataStoreUrl=my-zk:2181`? Since we're not requiring it even there.

Reply:
It's true, I'll add a couple of tests. it seems not covered at all now